client/attest.go

```diff
@@ -24,6 +27,39 @@ type AttestOpts struct {
 	// firmware event log, where PCRs 0-9 and 14 are often measured. If the two
 	// logs overlap, server-side verification using this library may fail.
 	CanonicalEventLog []byte
+	// Indicates whether the AK certificate chain should be retrieved for validation.
+	ValidateCertChain bool
+}
+
+// Constructs the certificate chain for the key's certificate, using the provided HTTP client.
+func (k *Key) getCertificateChain() ([][]byte, error) {
```

Comment:
So this seems like it would just get the first step in the cert chain, but not necessarily _all_ steps in the cert chain.

For example, if our Leaf cert contains one IssuingCertificateURL, we will just fetch that single cert, but we won't do this process recursively. What we really need to do is the following:

For each step in our cert chain:
  - Make sure there aren't too many IssuingCertificateURLs (3 max is fine).
  - For each URL in IssuingCertificateURL:
    - Try to `Get` the payload
    - Try to parse the cert
    - Check if the cert we got is what we expect (i.e. it signs the current cert in our chain)
    - If this doesn't work, move on to next url
  - append the cert to the chain
  - Take the cert we just appened in the previous step and repeat the process 
  - Cap the total depth we can go (4 max is fine)

Reply:
If, at any depth, we go through a list of IssuingCertificateURLs and can't find the expected one, should it return an error or just provide the certificates so far (if any).

Reply:
IIUC, we should return an error if we cannot find a root cert. (Or it'll be checked elsewhere?)

Reply:
The root cert should be checked by the attestation service when it verifies the attestation, we aren't expecting to check it here.